src/borg/remote.py

```diff
@@ -53,6 +55,50 @@ class UnexpectedRPCDataFormatFromServer(Error):
     """Got unexpected RPC data format from server."""
 
 
+# Protocol compatibility:
+# In general the server is responsible for rejecting too old clients and the client it responsible for rejecting
+# too old servers. This ensures that the knowledge what is compatible is always held by the newer component.
+#
+# The server can do checks for the client version in RepositoryServer.negotiate. If the client_data is 2 then
+# client is in the version range [0.29.0, 1.0.6] inclusive. For newer clients client_data is a dict which contains
+# client_version.
+#
+# For the client the return of the negotiate method is either 2 if the server is in the version range [0.29.0, 1.0.6]
+# inclusive, or it is a dict which includes the server version.
+#
+# All method calls on the remote repository object must be whitelisted in RepositoryServer.rpc_methods and have api
+# stubs in RemoteRepository. The @api decorator on these stubs is used to set server version requirements.
+#
+# Method parameters are identfied only by name and never by position. Unknown parameters are ignored by the server side.
+# If a new parameter is important and may not be ignored, on the client a parameter specific version requirement needs
+# to be added.
+# When parameters are removed, they need to be preserved  as defaulted parameters on the client stubs so that older
+# servers still get the from them still needed parameters.
+
+
+compatMap = {
+    'check' : ('repair', 'save_space'],
+    'commit': ('save_space'],
+    'rollback': (),
+    'destroy': (),
+    '__len__': (),
+    'list': ('limit', 'marker'],
+    'put': ('id_', 'data'],
+    'get': ('id_'],
+    'delete': ('id_'],
```

Comment:
id is definitely prettier, but might cause some complaint from linters because it shadows a builtin.

if we want to avoid that, I'ld rather use a more specific word than just appending a underscore. most such "shadows builtin" issues can be fixed by being more specific. So, not "id", but "obj_id" or "oid" or ...

otoh, we could also switch off the linter warning.